code/game/turfs/space/space.dm

```diff
@@ -182,3 +182,11 @@
 
 /turf/open/space/acid_act(acidpwr, acid_volume)
 	return 0
+
+/turf/open/space/rcd_act(mob/user, obj/item/weapon/rcd/the_rcd)
+	switch(the_rcd.mode)
+		if(RCD_FLOORWALL)
```

Comment:
I mean you're kind of copying this mode switching thing all over; all you need to do is have a proc like `can_rcd(mob/user, obj/item/weapon/rcd/the_rcd)` that returns a list with assoc values that you check, and any other checks you need.
Like the clockwork proselytizer, in fact. Look at that and do what it does for RCDs instead.